iml-manager-cli/src/main.rs

```diff
@@ -18,9 +18,101 @@ pub enum StratagemCommand {
         /// The name of the filesystem to scan
         #[structopt(short = "fs", long = "filesystem")]
         fs: String,
+        /// The report duration
+        #[structopt(short = "rd", long = "report", parse(try_from_str = "parse_duration"))]
+        rd: Option<u32>,
+        /// The purge duration
+        #[structopt(short = "pd", long = "purge", parse(try_from_str = "parse_duration"))]
+        pd: Option<u32>,
     },
 }
 
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+struct StratagemCommandData {
+    filesystem: String,
+    report_duration: Option<u32>,
+    purge_duration: Option<u32>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum DurationErrorKind {
+    /// No unit provided
+    NoUnit,
+    /// Invalid Unit provided
+    InvalidUnit,
+    /// Invalid value
+    InvalidValue,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+struct DurationParseError {
+    kind: DurationErrorKind,
+}
+
+impl DurationParseError {
+    pub fn to_string(&self) -> String {
+        match self.kind {
+            DurationErrorKind::NoUnit => String::from("No unit was specified."),
+            DurationErrorKind::InvalidUnit => {
+                String::from("Invalid unit provided. Valid units include: h, d")
+            }
+            DurationErrorKind::InvalidValue => String::from(
+                "Invalid value provided. Duration must be in the form of '<u32 value>{h|d}'",
+            ),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum DurationUnit {
+    /// Duration in hours
+    Hours,
+    /// Duration in days
+    Days,
+}
+
+fn parse_duration(src: &str) -> Result<u32, DurationParseError> {
+    if src.len() < 2 {
+        return Err(DurationParseError {
+            kind: DurationErrorKind::InvalidValue,
+        });
+    }
+
+    let val = &src[0..src.len() - 1];
+    let unit = &src[src.len() - 1..];
+
+    let original_val: u32;
+    if let Ok(i) = val.parse::<u32>() {
+        original_val = i
+    } else {
+        return Err(DurationParseError {
+            kind: DurationErrorKind::InvalidValue,
+        });
+    }
+
+    let duration_unit: DurationUnit = match unit.chars().last() {
+        Some('h') => DurationUnit::Hours,
+        Some('d') => DurationUnit::Days,
+        Some('1'...'9') => {
+            return Err(DurationParseError {
+                kind: DurationErrorKind::NoUnit,
+            })
+        }
+        _ => {
+            return Err(DurationParseError {
+                kind: DurationErrorKind::InvalidUnit,
+            })
+        }
+    };
+
+    let interval = match (original_val, duration_unit) {
+        (original_value, DurationUnit::Hours) => original_value * 3600,
+        (original_value, DurationUnit::Days) => original_value * 86400,
+    };
+
+    Ok(interval)
```

Comment:
There is no need for a `DurationUnit` enum when you are only using it for an alias on the match above.

The above can be simplified to:

```rust
    match unit {
        Some('h') => Ok(val * 3_600),
        Some('d') => Ok(val * 86_400),
        Some('1'...'9') => Err(DurationParseError {
            kind: DurationErrorKind::NoUnit,
        }),
        _ => Err(DurationParseError {
            kind: DurationErrorKind::InvalidUnit,
        }),
    }
```